st2actions/st2actions/runners/ssh/paramiko_ssh.py

```diff
@@ -560,6 +560,24 @@ def _connect(self, host, socket=None):
         if socket:
             conninfo['sock'] = socket
 
+        if cfg.CONF.ssh_runner.use_ssh_config:
+            ssh_config = paramiko.SSHConfig()
+            user_config_file = os.path.expanduser("~/.ssh/config")
```

Comment:
Just noting that the actionrunner would run as root if people use PAM authentication as the auth backend. In this case, expanduser for ~ will be /root. Usually, actionrunner process runs as ``st2`` in which case ~ would be /home/st2/. I honestly prefer to pass the full path to ssh config file via config. Also, log an exception with appropriate error message on lack of permissions to read would be critical. 

I also think in terms of code, you could stub it out into a method so it's more unit testable. So lines 564 - 579 could be a method _get_conninfo_from_ssh_config() and this method could return a conninfo dict. 

Thoughts? I am on slack if you want to talk this out. stackstorm.com/community-signup